src/v2/view-builder/views/consent/EmailMagicLinkOTPTerminalView.js

```diff
@@ -1,19 +1,44 @@
 import { loc, View } from 'okta';
 import hbs from 'handlebars-inline-precompile';
 
+const generateGeolocationString = (location) => {
+  let geolocation = null;
+  let values = [];
+  if (location) {
+    // see which info is provided
+    if (location.city) {
+      values.push(location.city);
+    }
+    if (location.state) {
+      values.push(location.state);
+    }
+    if (location.country) {
+      values.push(location.country);
+    }
+    switch (values.length) {
+    case 0:
+      break;
+    case 1:
+      geolocation = values[0];
+      break;
+    case 2:
+      geolocation = loc('idx.return.link.otponly.city.state.country.some', 'login', values);
+      break;
+    case 3:
+      geolocation = loc('idx.return.link.otponly.city.state.country.all', 'login', values);
```

Comment:
I doubt customers will ever want to override these keys, but can we generalize these to avoid the long names?

I'd suggest:

```suggestion
      geolocation = loc('geolocation.formatting.partial', 'login', values);
      break;
    case 3:
      geolocation = loc('geolocation.formatting.full', 'login', values);
```